docker/images/ckan/ckan-entrypoint.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -u
+
+check_port() {
+    local host="$1"
+    local port="$2"
+
+    echo "Checking service availability at ${host}:${port}..."
+
+    while true; do
+        # Use nmap to scan the specified port on the given host
+        nmap -p "${port}" "${host}" | grep -q "open"
+
+        # If the port is open, exit the loop
+        if [ $? -eq 0 ]; then
+            echo "Server is available at ${host}:${port}"
+            break
+        else
+            echo "Server at ${host}:${port} is not available. Retrying in 1 seconds..."
+            sleep 1
+        fi
+    done
+}
+
+## Check DB up
+check_port db 5432
+
+## Check Solr up
+check_port solr 8983
+
+## Check Redis up
+check_port redis 6379
+
+## Initialize CKAN config, database and admin account
+CKAN_CONFIG_FILE=/etc/ckan/default/ckan.ini
+CKAN_INIT_STATUS_FILE=/etc/ckan/default/.ckan_initialized
+
+if [ "$CD2_HOST_PORT" = "443" ]
+then export CKAN_SITE_URL="https://$CD2_HOST"
+else export CKAN_SITE_URL="https://$CD2_HOST:$CD2_HOST_PORT"
+fi
+
+if test -f "$CKAN_INIT_STATUS_FILE"
+then echo "Configuration and database already initialized."
+else echo "Initializing configuration ..."
+     export BEAKER_SESSION_SECRET=$(openssl rand -base64 32)
+     export SECRET_TOKEN_VALUE=$(openssl rand -base64 32)
+     export APP_INSTANCE_UUID=$(uuidgen --name "$CD2_HOST" --namespace "@url" --sha1)
+     perl -pi.bak -e '$beaker_session_secret=$ENV{BEAKER_SESSION_SECRET}; s/BEAKER_SESSION_SECRET/$beaker_session_secret/ge' "$CKAN_CONFIG_FILE"
+     perl -pi.bak -e '$secret_token=$ENV{SECRET_TOKEN_VALUE}; s/SECRET_TOKEN_VALUE/$secret_token/ge' "$CKAN_CONFIG_FILE"
+     perl -pi.bak -e '$app_instance_uuid=$ENV{APP_INSTANCE_UUID}; s/APP_INSTANCE_UUID/$app_instance_uuid/ge' "$CKAN_CONFIG_FILE"
+     perl -pi.bak -e '$ckan_database_password=$ENV{CKAN_DATABASE_PASSWORD}; s/CKAN_DATABASE_PASSWORD/$ckan_database_password/ge' "$CKAN_CONFIG_FILE"
+     perl -pi.bak -e '$ckan_site_url=$ENV{CKAN_SITE_URL}; s/CKAN_SITE_URL/$ckan_site_url/ge' "$CKAN_CONFIG_FILE"
+     echo "Initializing database ..."
+     /usr/lib/ckan/default/bin/ckan -c "$CKAN_CONFIG_FILE" db init
+     /usr/lib/ckan/default/bin/ckan -c "$CKAN_CONFIG_FILE" user add ckanadmin password="$CKAN_ADMIN_PASSWORD" email=ckanadmin@localhost name=ckanadmin
+     /usr/lib/ckan/default/bin/ckan -c "$CKAN_CONFIG_FILE" sysadmin add ckanadmin
+     sudo chown ckan /ckan_api_key
```

Comment:
This line can be removed, since the API key volume is not needed for the CD2 catalog